bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/SAICiSMARTBridgeHandler.java

```diff
@@ -0,0 +1,298 @@
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.bn.coders.IASN1PreparedElement;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.GsonBuilder;
+
+import net.heberling.ismart.asn1.v1_1.Message;
+import net.heberling.ismart.asn1.v1_1.MessageCoder;
+import net.heberling.ismart.asn1.v1_1.entity.AlarmSwitch;
+import net.heberling.ismart.asn1.v1_1.entity.AlarmSwitchReq;
+import net.heberling.ismart.asn1.v1_1.entity.MP_AlarmSettingType;
+import net.heberling.ismart.asn1.v1_1.entity.MP_UserLoggingInReq;
+import net.heberling.ismart.asn1.v1_1.entity.MP_UserLoggingInResp;
+import net.heberling.ismart.asn1.v1_1.entity.MessageListReq;
+import net.heberling.ismart.asn1.v1_1.entity.MessageListResp;
+import net.heberling.ismart.asn1.v1_1.entity.StartEndNumber;
+import net.heberling.ismart.asn1.v1_1.entity.VinInfo;
+
+/**
+ * The {@link SAICiSMARTBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Heberling - Initial contribution
+ */
+@NonNullByDefault
+public class SAICiSMARTBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SAICiSMARTBridgeHandler.class);
+
+    private @Nullable SAICiSMARTBridgeConfiguration config;
+
+    private @Nullable String uid;
+
+    private @Nullable String token;
+
+    private @Nullable Collection<VinInfo> vinList;
+    private HttpClientFactory httpClientFactory;
+    private @Nullable Future<?> pollingJob;
+
+    public SAICiSMARTBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
```

Comment:
It might be better to cache the httpclient instead of the factory.
Atleast don't create a new client  (by calling `httpClientFactory.getCommonHttpClient();` every request.